dd-sdk-android/src/main/kotlin/com/datadog/android/rum/internal/anr/ANRDetectorRunnable.kt

```diff
@@ -0,0 +1,89 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.rum.internal.anr
+
+import android.os.Handler
+import com.datadog.android.rum.GlobalRum
+import com.datadog.android.rum.RumErrorSource
+
+/**
+ * A Runnable running on a background thread detecting ANR on the main thread.
+ *
+ * It runs in a background thread and schedules regular no-op
+ */
+internal class ANRDetectorRunnable(
+    private val handler: Handler,
+    private val anrThresholdMs: Long = ANR_THRESHOLD_MS,
+    private val anrTestDelayMs: Long = ANR_TEST_DELAY_MS
+) : Runnable {
+
+    private var shouldStop = false
+
+    // region Runnable
+
+    override fun run() {
+
+        while (!Thread.interrupted()) {
+            if (shouldStop) return
+
+            try {
+                val callback = CallbackRunnable()
+
+                synchronized(callback) {
+                    handler.post(callback)
+                    callback.wait(anrThresholdMs)
+
+                    if (!callback.wasCalled()) {
+                        GlobalRum.get().addError(
+                            ANR_MESSAGE,
+                            RumErrorSource.SOURCE,
+                            ANRException(handler.looper.thread),
+                            emptyMap()
+                        )
+                        callback.wait()
```

Comment:
Yes, that's the purpose. This will wait for the `CallbackRunnable` to be called on the main looper and ensure we don't report the same ANR more than once.

Reply:
I see, do you think we can add some unit tests for these scenarios ? Something like:
`we detected an ANR, now we put the thread to `wait` until the runnable is notified from the MainThread in order to schedule it again`